src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs

```diff
@@ -60,68 +60,74 @@ public ElasticsearchRequestPipelineDiagnosticListener(ElasticsearchClientInstrum
 
         public override void OnStartActivity(Activity activity, object payload)
         {
-            var uri = this.uriFetcher.Fetch(payload);
+            // By this time, samplers have already run and
+            // activity.IsAllDataRequested populated accordingly.
 
-            if (uri == null)
+            if (Sdk.SuppressInstrumentation)
             {
-                ElasticsearchInstrumentationEventSource.Log.NullPayload(nameof(ElasticsearchRequestPipelineDiagnosticListener), nameof(this.OnStartActivity));
                 return;
             }
 
-            ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
-            ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Client);
+            if (activity.IsAllDataRequested)
+            {
+                var uri = this.uriFetcher.Fetch(payload);
 
-            var method = this.methodFetcher.Fetch(payload);
-            activity.DisplayName = this.GetDisplayName(activity, method);
+                if (uri == null)
+                {
+                    ElasticsearchInstrumentationEventSource.Log.NullPayload(nameof(ElasticsearchRequestPipelineDiagnosticListener), nameof(this.OnStartActivity));
+                    return;
+                }
 
-            if (this.options.SuppressDownstreamInstrumentation)
-            {
-                SuppressInstrumentationScope.Enter();
-            }
+                ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
+                ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Client);
 
-            if (!activity.IsAllDataRequested)
-            {
-                return;
-            }
+                var method = this.methodFetcher.Fetch(payload);
+                activity.DisplayName = this.GetDisplayName(activity, method);
 
-            var elasticIndex = this.GetElasticIndex(uri);
-            activity.DisplayName = this.GetDisplayName(activity, method, elasticIndex);
-            activity.SetTag(SemanticConventions.AttributeDbSystem, DatabaseSystemName);
+                if (this.options.SuppressDownstreamInstrumentation)
+                {
+                    SuppressInstrumentationScope.Enter();
+                }
 
-            if (elasticIndex != null)
-            {
-                activity.SetTag(SemanticConventions.AttributeDbName, elasticIndex);
-            }
+                var elasticIndex = this.GetElasticIndex(uri);
+                activity.DisplayName = this.GetDisplayName(activity, method, elasticIndex);
```

Comment:
DisplayName was already set few lines above. One of them could be removed

Reply:
The reason it was set above was because that used to be outside of the check for `IsAllDataRequested` and gave the activity a basic display name and if `IsAllDataRequested` was true it set it to a more details display name.